agent/os_helper.py

```diff
@@ -0,0 +1,269 @@
+                if __file__.encode() in package_header[rpm.RPMTAG_FILENAMES]:
+                    return Installation.RPM
+    else:
+        cache = apt.Cache()
+        if __file__ in cache['wott-agent'].installed_files:
+            return Installation.DEB
+    # Other.
+    if isinstance(agent.__version__, pkg_resources.Distribution):
+        return Installation.PYTHON_PACKAGE
+    return Installation.NONE
+
+
+def get_packages():
+    # For apt-based distros.
+    try:
+        import apt
+    except ModuleNotFoundError:
+        # For rpm-based distros.
+        import rpm
+        ts = rpm.ts()
+        mi = ts.dbMatch()
+        packages = [package_header for package_header in mi]
+        packages_str = str(
+            sorted((package_header[rpm.RPMTAG_NAME].decode(), package_header[rpm.RPMTAG_EVR].decode())
+                   for package_header in packages))
+        packages_hash = hashlib.md5(packages_str.encode()).hexdigest()
+        return {
+            'hash': packages_hash,
+            'packages': [{
+                'name': package_header[rpm.RPMTAG_NAME].decode(),
+                'version': package_header[rpm.RPMTAG_EVR].decode(),
+                'arch': package_header[rpm.RPMTAG_ARCH].decode() if package_header[rpm.RPMTAG_ARCH] is not None
+                else 'noarch',
+                # Looks like there's no source name/version in the rpm package info.
+                # TEMP: pass package name and version.
+                'source_name': package_header[rpm.RPMTAG_NAME].decode(),
+                'source_version': package_header[rpm.RPMTAG_EVR].decode()
+            } for package_header in packages]
+        }
+    else:
+        cache = apt.Cache()
+        packages = [deb for deb in cache if deb.is_installed]
+        packages_str = str(sorted((deb.installed.package.name, deb.installed.version) for deb in packages))
+        packages_hash = hashlib.md5(packages_str.encode()).hexdigest()
+        return {
+            'hash': packages_hash,
+            'packages': [{
+                'name': deb.installed.package.name,
+                'version': deb.installed.version,
+                'arch': deb.installed.architecture,
+                'source_name': deb.installed.source_name,
+                'source_version': deb.installed.source_version
+            } for deb in packages]
+        }
+
+
+def get_os_release():
+    """
+    Returns a dict with the following items:
+    distro: Concrete distro name. Examples: raspbian, ubuntu, debian, ubuntu-core.
+    version: Short, numerical version. Examples: 9, 18.04, 18.
+    distro_root: The root distro (from which the distro was branched). Optional. Examples: debian.
+    full_version: Longer, human-readable version. Optional. Examples (last one is from ubuntu-core):
+        "9 (stretch)", "18.04.3 LTS (Bionic Beaver)", 18
+    codename: Distro version codename. Optional. Examples: stretch, bionic.
+    """
+
+    os_release = Path('/etc/os-release')
```

Comment:
Example of `/etc/os-release` on AWS Linux:


```
$ cat /etc/os-release
NAME="Amazon Linux"
VERSION="2"
ID="amzn"
ID_LIKE="centos rhel fedora"
VERSION_ID="2"
PRETTY_NAME="Amazon Linux 2"
ANSI_COLOR="0;33"
CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2"
HOME_URL="https://amazonlinux.com/"
```

Reply:
and?

Reply:
And nothing. Just as a reference.